### PR TITLE
Fix base_unittests and run passing tests on win32

### PR DESCRIPTION
**Pull request by andrewsavage1**
b/330792170

### CODE REVIEW

File: base/files/file_unittest.cc [hunk 1]
```diff
@@ -97,6 +98,7 @@ TEST(FileTest, Create) {
     EXPECT_EQ(base::File::FILE_ERROR_EXISTS, file.error_details());
```
Comment (johnxwork) [inline base/files/file_unittest.cc:98]: Do you know why did this part fail?

Comment (johnxwork) [reply base/files/file_unittest.cc:98]: Was it also because "win32 SbFileOpen doesn't fail on relative path"?

**Review by johnxwork — APPROVED** [review]
We can modify the commit/PR headline to "fix base_unittests on win32"

File: base/files/file_unittest.cc (hunk 1 above)
Comment (andrewsavage1) [reply base/files/file_unittest.cc:98]: I didn't look much into it and just took what was from trunk, but it looked to me like differences in our file opening error codes from base's

**Pull request merged**